### PR TITLE
add a new action to merge json

### PR DESCRIPTION
**Pull request by punkle**
add a new action to merge json

#### :heavy_check_mark: Checklist

- [x] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/json/merge.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+      let existingContent = {};
+
+      if (fs.existsSync(sourceFilepath)) {
+        existingContent = JSON.parse(
+          fs.readFileSync(sourceFilepath).toString(),
+        );
+      }
+
+      fs.writeFileSync(
```
Comment (punkle) [inline plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/json/merge.ts:67]: @iain-b as discussed.

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/json/merge.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    async handler(ctx) {
+      const sourceFilepath = resolveSafeChildPath(
+        ctx.workspacePath,
+        ctx.input.path,
+      );
+
+      let existingContent = {};
+
+      if (fs.existsSync(sourceFilepath)) {
```
Comment (punkle) [inline plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/json/merge.ts:61]: I think this adds a helpful, add or append type behaviour.

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/json/merge.ts (hunk 1 above)
Comment (iain-b) [reply plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/json/merge.ts:67]: Yep, lgtm

**Review by iain-b — APPROVED** [review]

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/json/merge.ts (hunk 1 above)
Comment (iain-b) [reply plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/json/merge.ts:67]: down the line other file formats might be handy but I'd expect json would be the most common use case for this.

**Pull request merged**